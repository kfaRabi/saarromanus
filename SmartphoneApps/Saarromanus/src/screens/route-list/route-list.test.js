import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import RouteListScreen from './index';

const mockStore = configureStore();

const mockFn = jest.fn();
const navigation = {
	navigate: mockFn,
};

const initialState = {
	config: { checkForUpdate: false },
	routes: [
		{
			id: 'KBaeZFpmqRafsXhcsq3V',
			en: {
				name: 'Dillingen Route',
				information: 'Info..',
			},
			de: {
				name: 'Dillingen Route',
				information: 'Info..',
			},
			fr: {
				name: 'Delingen Route',
				information: 'Info..',
			},
		},
		{
			id: 'lzOL126R7Em8FDO0jlHH',
			en: {
				name: 'Saarland Route',
				information: 'Information',
			},
			de: {
				name: 'Saarland Route',
				information: 'Informationen',
			},
			fr: {
				information: 'Informations',
				name: 'Route de la Sarre',
			},
		},
	],
};

let wrapper;
let store;
let component;

beforeEach(() => {
	store = mockStore(initialState);
	wrapper = shallow(
		<RouteListScreen store={store} navigation={navigation} />
	).dive();
	component = wrapper.dive();
});

describe('<RouteListScreen />', () => {
	it('renders properly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('has one main <View /> component', () => {
		expect(component.find(`[data-test-id='mainComponent']`).length).toBe(1);
	});

	describe('Routes List', () => {
		it('length should not be zero when there are routes', () => {
			expect(component.find(`[data-test-id='listItem']`).length).not.toBe(
				0
			);
		});
		it('length should be equal to the number of routes received from the store', () => {
			expect(component.find(`[data-test-id='listItem']`).length).toBe(
				initialState.routes.length
			);
		});
		describe('Single Route', () => {
			it('should navigate to the Route Details page when pressed/tapped', () => {
				component
					.find(`[data-test-id='listItem']`)
					.find('TouchableOpacity')
					.forEach(node => {
						node.simulate('press');
					});
				expect(mockFn).toBeCalled();
				expect(mockFn.mock.calls.length).toBe(2);
			});
			it('should not fire navigation event if it is not pressed/tapped', () => {
				expect(jest.fn()).not.toBeCalled();
				expect(jest.fn().mock.calls.length).toBe(0);
			});
		});
	});
});
