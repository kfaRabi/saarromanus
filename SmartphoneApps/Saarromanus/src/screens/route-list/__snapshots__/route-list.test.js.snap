// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteListScreen /> renders properly 1`] = `
<RouteListScreen
  checkUpdateStatus={false}
  getRouteList={[Function]}
  navigation={
    Object {
      "navigate": [MockFunction],
    }
  }
  populateRouteList={[Function]}
  routes={
    Array [
      Object {
        "de": Object {
          "information": "Info..",
          "name": "Dillingen Route",
        },
        "en": Object {
          "information": "Info..",
          "name": "Dillingen Route",
        },
        "fr": Object {
          "information": "Info..",
          "name": "Delingen Route",
        },
        "id": "KBaeZFpmqRafsXhcsq3V",
      },
      Object {
        "de": Object {
          "information": "Informationen",
          "name": "Saarland Route",
        },
        "en": Object {
          "information": "Information",
          "name": "Saarland Route",
        },
        "fr": Object {
          "information": "Informations",
          "name": "Route de la Sarre",
        },
        "id": "lzOL126R7Em8FDO0jlHH",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
